blockstore/apps/api/v1/serializers/bundles.py

```diff
@@ -140,33 +133,13 @@ def _expand_url(self, url):
 
         def to_representation(self, value):
             """Snapshot JSON serialization."""
-            snapshot = value
-            snapshot_repo = SnapshotRepo()
-            info = {
-                'hash_digest': snapshot.hash_digest.hex(),
-                'created_at': snapshot.created_at,
-            }
-
-            info['files'] = {
-                path: {
-                    "url": self._expand_url(snapshot_repo.url(snapshot, path)),
-                    "size": file_info.size,
-                    "hash_digest": file_info.hash_digest.hex(),
-                }
-                for path, file_info in snapshot.files.items()
-            }
-
-            info['links'] = {
-                link.name: {
-                    "direct": self._serialized_dep(link.direct_dependency),
-                    "indirect": [
-                        self._serialized_dep(dep)
-                        for dep in link.indirect_dependencies
-                    ]
-                }
-                for link in snapshot.links
-            }
-
+            view_kwargs = self.context['view'].kwargs
+            info = value.serialize(context=self.context)
+            for file_name, file_info in info['files'].items():
+                file_info['url'] = self._expand_url('{}?file={}'.format(
+                    reverse('api:v1:bundleversion-redirect', kwargs=view_kwargs),
+                    file_name,
+                ))
```

Comment:
Do I understand correctly that this will always generate a freshly signed URL, when using the S3 backend?

Reply:
yep, that's the idea

Reply:
Isn't that going to make caching performance worse though (always generating a freshly signed URL)?

Before, we'd generate a signed URL then cache it for an hour (or whatever the URLs expiry time was). During that hour, browsers wouldn't try to re-load the asset, they'd just serve it from cache. This should have been working well except for some cache invalidation bug somewhere.

Now, if the URL is changing every time, browsers will never serve the asset from cache. Right?